src/dropdown.ts

```diff
@@ -1848,30 +1916,89 @@ export default class Dropdown extends LitElement implements FormControl {
       for (const option of this.#optionElements) {
         option.hidden = !options.includes(option);
       }
+    }
 
-      const firstVisibleOption = this.#optionElementsNotHidden?.at(0);
+    this.isCommunicateItemCountToScreenreaders = true;
 
-      // When filtering filters out the active option, make the first option active
-      // if there is one.
-      if (firstVisibleOption && this.activeOption?.hidden) {
-        this.activeOption.privateActive = false;
-        this.#previouslyActiveOption = firstVisibleOption;
-        this.ariaActivedescendant = firstVisibleOption.id;
+    if (this.#optionElementsNotHidden) {
+      this.itemCount = this.#optionElementsNotHidden.length;
```

Comment:
Same as before. Just moved it up so all the active option setting conditions are next to each other.